### PR TITLE
Add regression test for client scope issue

### PR DESCRIPTION
**Pull request by jacquelinekay**
Connects to ros2/rclcpp#192

### CODE REVIEW

File: test_rclcpp/test/test_client_scope_client.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include "gtest/gtest.h"
+#include "rclcpp/rclcpp.hpp"
+#include <memory>
+#include "test_rclcpp/srv/add_two_ints.hpp"
+
+TEST(service_client, client_scope_regression_test) {
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_client_scope_client.cpp]: Needs rmw specific class name.

File: test_rclcpp/test/test_client_scope_client.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  auto node = rclcpp::node::Node::make_shared("client_scope_regression_test");
+
+  //Extra scope so the first client will be deleted afterwards
+  {
+    std::cout << "Creating first client" << std::endl;
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_client_scope_client.cpp]: `printf`

**Comment by dirk-thomas** [comment]
Squashing in preparation of the merge.

**Comment by dirk-thomas** [comment]
@jacquelinekay I added plenty of prints to see the progress: 8856aefe2efa860a63f09ad80f547fe1a0626768

**Comment by gerkey** [comment]
+1. This test has turned out to be super useful.

**Pull request merged**